pubtools/pulplib/_impl/fake/controller.py

```diff
@@ -98,3 +100,19 @@ def upload_history(self):
                 checksum of the file uploaded
         """
         return self.client._upload_history[:]
+
+    def update_maintenance_file(self):
```

Comment:
Why is the behavior hardcoded like this?  I suggest the way it should work is:

- as mentioned in earlier comment, there should be some proper object (not raw dict) modelling the maintenance status
- then you could have set_maintenance(x) which simply stores x as the new maintenance status

Also, this is a public method and should be documented.

Reply:
set_maintenance is supposed to be done in pubtools.pulp, I'm not sure how to do it here?

Reply:
That's actually a good point as well and something I missed:

I think you should also have the logic for setting maintenance mode belonging to this library.  In pubtools-pulp, it would mainly be just calling a method or two to get/set the mode according to arguments.  That would ensure all the details about e.g. which repo contains that file, what's the format of the file, are all encapsulated in this library and not presented elsewhere.

Reply:
why is this method implemented only in fake client?

Reply:
Hm, this is in the controller, not client.